### PR TITLE
#165250363 User should be able to get all report categories

### PR DESCRIPTION
**Pull request by chukwudidu**
#### Description
Creates the endpoint that gives a user the ability to get all report categories.
#### Type of change
- [x]  Non-breaking change(introduces new endpoint to codebase)
#### How Has This Been Tested?
- All tests are currently passing
- It was also tested on Postman

##### Screenshots
Screenshot 2019-04-11 at 12 03 52 PM

#### Checklist:
- [x]  Implement get all report categories method
#### PIVOTAL BOARD
[#165250363](https://www.pivotaltracker.com/story/show/165250363)

### CODE REVIEW

**Review by davidokonji — COMMENTED** [review]
Good work done

File: server/routes/index.js [hunk 1]
```diff
@@ -2,12 +2,14 @@ import express from 'express';
...
 import userRoutes from './users.routes';
 import articleRoutes from './article.routes';
+import reportCategories from './report-categories.routes';
 
 const router = express.Router();
 
 router.use('/auth', authRoutes);
 router.use('/users', userRoutes);
 router.use('/article', articleRoutes);
+router.use('/reportCategories', reportCategories);
```
Comment (Chiazokam) [inline server/routes/index.js]: Hi @cupidcoder I think this can be better presented with `/report/categories`

**Review by Chiazokam — CHANGES_REQUESTED** [review]

**Review by samorahno — COMMENTED** [review]
Good job @cupidcoder

File: server/routes/index.js (hunk 1 above)
Comment (chukwudidu) [reply server/routes/index.js]: @Chiazokam I agree, I would implement right away

File: server/models/reporttype.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      allowNull: false,
+    },
+    name: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    }
+  }, {});
+  ReportType.associate = function (models) { };
+  return ReportType;
+};
```
Comment (TheDrizzyWay) [inline server/models/reporttype.js:23]: Good job all round. Do you plan to add associations to the model later on?

Comment (chukwudidu) [reply server/models/reporttype.js:23]: @TheDrizzyWay Thanks. Yes, I actually added associations to `Report` in my `report article feature`. A `reportType` has many `reports`

**Review by geekdanitek — APPROVED** [review]

**Review by samorahno — APPROVED** [review]

**Review by BukkyOmo — APPROVED** [review]

File: server/routes/index.js (hunk 1 above)
Comment (paradisekelechi) [reply server/routes/index.js]: Did you implement this @cupidcoder ?

Comment (chukwudidu) [reply server/routes/index.js]: @Chiazokam @paradisekelechi I have done this. Thanks

**Review by Chiazokam — APPROVED** [review]

**Pull request merged**